scipy/signal/signaltools.py

```diff
@@ -2971,13 +2971,13 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=None):
     if ftype == 'fir':
         if n is None:
             n = 30
-        system = lti(firwin(n+1, 1. / q, window='hamming'), 1.)
+        system = dlti(firwin(n+1, 1. / q, window='hamming'), 1.)
     elif ftype == 'iir':
         if n is None:
             n = 8
-        system = lti(*cheby1(n, 0.05, 0.8 / q))
-    elif isinstance(ftype, lti):
-        system = ftype
+        system = dlti(*cheby1(n, 0.05, 0.8 / q))
+    elif isinstance(ftype, dlti):
+        system = ftype._as_tf()
```

Comment:
This method is a little cleaner than `to_tf`, since it doesn't copy if the system is already in TF form. I suppose there can be a separate discussion if we want public `to_foo` and `as_foo` methods in the LTI system objects.